libs/island-ui/core/src/lib/InputFileUpload/InputFileUpload.tsx

```diff
@@ -322,9 +322,11 @@ export const InputFileUpload = ({
       <Text variant="h4">{header}</Text>
       <Text>{description}</Text>
       <Box marginY={4}>
-        <Button variant="ghost" icon="attach" disabled={disabled}>
-          {buttonLabel}
-        </Button>
+        {(multiple || (!multiple && fileList.length === 0)) && (
+          <Button variant="ghost" icon="attach" disabled={disabled}>
+            {buttonLabel}
+          </Button>
+        )}
```

Comment:
Maybe we should let the condition wrap the parent `<Box>` so that unnecessary margin is avoided

Reply:
I want the space, this is a bit cramped without it. I thought this was a bit more clean than adding a space on this condition as well